gammapy/scripts/image_pipe.py

```diff
@@ -2,56 +2,160 @@
+        Energy band for which we want to compute the image
+    offset_band : `astropy.coordinates.Angle`
+        Offset Band where you compute the image
+    data_store : `~gammapy.data.DataStore`
+        `DataStore` where are situated the events
+    obs_table : `~astropy.table.Table`
+            Required columns: OBS_ID
+    exclusion_mask : `~gammapy.image.ExclusionMask`
+            Exclusion regions
     """
 
-    def __init__(self, config):
-        self.config = config
-
-    @classmethod
-    def from_yaml(cls, filename):
-        """Read config from YAML file."""
-        import yaml
-        log.info('Reading {}'.format(filename))
-        with open(filename) as fh:
-            config = yaml.safe_load(fh)
-        return cls(config)
-
-    def run(self):
-        """Run analysis chain."""
-        log.info('Running analysis ...')
-        print(self.config['general']['outdir'])
-        print(self.config)
-
-    def make_counts(self):
+    def __init__(self, empty_image=None,
+                 energy_band=None, offset_band=None,
+                 data_store=None, obs_table=None, exclusion_mask=None):
+        self.maps = SkyMapCollection()
+
+        self.data_store = data_store
+        if not obs_table:
+            self.obs_table = self.data_store.obs_table
+        else:
+            self.obs_table = obs_table
+        self.energy_band = energy_band
+        self.offset_band = offset_band
+
+        self.empty_image = empty_image
+        self.header = self.empty_image.to_image_hdu().header
+        if exclusion_mask:
+            self.maps['exclusion'] = exclusion_mask
+
+    def make_counts(self, obs_id):
+        """Fill the counts image for the events of one observation.
+
+        Parameters
+        ----------
+        obs_id : int
+            Number of the observation
+
+        """
+        self.maps["counts"] = SkyMap.empty_like(self.empty_image)
+        obs = self.data_store.obs(obs_id=obs_id)
+        events = obs.events.select_energy(self.energy_band)
+        events = events.select_offset(self.offset_band)
+        self.maps["counts"].fill(events=events)
+        self.maps["counts"].data = self.maps["counts"].data.value
         log.info('Making counts image ...')
 
-    def make_exposure(self):
-        log.info('Making exposure image ...')
+    def make_total_counts(self):
+        """Stack the total count from the observation in the 'DataStore'
+
+        """
+        self.maps["total_counts"] = SkyMap.empty_like(self.empty_image)
+        for obs_id in self.obs_table['OBS_ID']:
+            self.make_counts(obs_id)
+            self.maps["total_counts"].data += self.maps["counts"].data
+
+    def make_background(self, obs_id):
+        """Make the background map for one observation from a bkg model
+
+        Parameters
+        ----------
+        obs_id : int
+            Number of the observation
+
+        """
+        self.maps["bkg"] = SkyMap.empty_like(self.empty_image)
```

Comment:
Again, I would suggest to always do the computation first using a local variable, and to only store it as a data member on the last line.
I find this a little simpler, and also, if there is some error, it leaves the object in an unmodified state, which is better.